src/App.js

```diff
@@ -9,13 +9,137 @@ import Option from './Components/Option/Option';
 import './Css/Todo.css';
 import './App.css';
 
-class App extends Component { 
+const isNotCheckedAll = (todos = []) => todos.find(todo => !todo.isCompleted);
+
+const filterByStatus = (todos = [], status = '', id = '') => {
+  switch (status) {
+    case 'ACTIVE':
+      return todos.filter(todo => !todo.isCompleted);
+    case 'COMPLETED':
+      return todos.filter(todo => todo.isCompleted);
+    case 'REMOVE':
+      return todos.filter(todo => todo.id !== id);
+    default:
+      return todos;
+  }
+}
+class App extends PureComponent {
+  state = {
+    todoList: [],
+    todoEditingId: '',
+    isCheckedAll: false,
+    status: 'ALL',
+    itemLeft: 0,
+    shouldShowClearCompleted: false
+  }
+
+  componentWillMount() {
+    this.setState({
+      isCheckedAll: isNotCheckedAll(this.state.todoList)
```

Comment:
should not check logic (or api,...) in `componentWillMount`. It will be delayed the UI rendering